Topics/Tech_Stacks/JsonParsing.md

```diff
@@ -0,0 +1,231 @@
+# Introduction to JSON Parsing
+
+
+
+## Introduction to JSON
+
+JSON (JavaScript Object Notation), is a lightweight and human-readable data interchange format. It serves as a standard data format for transmitting and exchanging data between a server and a web application, as well as between different parts of an application. JSON is language-agnostic, meaning it can be easily understood and used by various programming languages.
+
+
+## JSON Data Types
+
+### Primitive Data Types
+
+JSON supports several primitive data types. These primitive data types are the basic building blocks used to represent values within a JSON structure. The primary primitive data types in JSON are: 
+
+ * String: Represents a sequence of characters enclosed in double quotation marks (")
+  
+    Example: "Hello, World!" 
+
+
+ * Number: Represents numeric values, including integers and floating-point numbers.
+    
+    Examples: 42, 3.14, -17 
+
+ * Boolean: Represents a logical value, either true or false. 
+    
+    Examples: true, false 
+
+ * Null: Represents an empty value or the absence of a value. 
+    
+    Example: null
+
+These primitive data types can be used alone or combined to create more complex JSON structures such as objects and arrays. For example, an object may contain key-value pairs where the values can be strings, numbers, booleans, null, or even nested objects and arrays.
+
+### Complex Data Types
+
+JSON allows for the construction of more complex data structures beyond primitive data types by using objects and arrays. 
+
+ * Objects: An object in JSON is an unordered collection of key-value pairs. Key-value pairs are separated by commas and enclosed in curly braces {}. Keys must be strings, and values can be strings, numbers, booleans, null, objects, or arrays. 
+ Example:
+
+
+{
+  "name": "John Doe",
+  "age": 30,
+  "isStudent": false,
+  "address": {
+    "city": "Exampleville",
+    "zipcode": "12345"
+  }
+}
+
+ * Array: An array in JSON is an ordered list of values. Values are separated by commas and enclosed in square brackets []. Values can be strings, numbers, booleans, null, objects, or other arrays. 
+ Example:
+
+ [
+  "apple",
+  "banana",
+  "orange",
+  {
+    "color": "red",
+    "quantity": 5
+  }
+]
```

Comment:
Please put in code block for better readability

Reply:
done.